src/gui/userstatus.cpp

```diff
@@ -28,39 +28,41 @@ namespace OCC {
 
 Q_LOGGING_CATEGORY(lcUserStatus, "nextcloud.gui.userstatus", QtInfoMsg)
 
-UserStatus::UserStatus(QObject *parent)
-    : QObject(parent)
-{
-}
-
-UserStatus::Status UserStatus::stringToEnum(const QString &status) const 
-{
-    // it needs to match the Status enum
-    const QHash<QString, Status> preDefinedStatus{{"online", Status::Online},
-                                               {"dnd", Status::DoNotDisturb}, //DoNotDisturb
-                                               {"away", Status::Away},
-                                               {"offline", Status::Offline},
-                                               {"invisible", Status::Invisible}};
+namespace {
+    UserStatus::Status stringToEnum(const QString &status) 
+    {
+        // it needs to match the Status enum
+        const QHash<QString, UserStatus::Status> preDefinedStatus{{"online", UserStatus::Status::Online},
+            {"dnd", UserStatus::Status::DoNotDisturb}, //DoNotDisturb
+            {"away", UserStatus::Status::Away},
+            {"offline", UserStatus::Status::Offline},
+            {"invisible", UserStatus::Status::Invisible}};
+        
+        // api should return invisible, dnd,... toLower() it is to make sure 
+        // it matches _preDefinedStatus, otherwise the default is online (0)
+        const auto statusKey = status.isEmpty() ? QStringLiteral("online") : status.toLower();
+        return preDefinedStatus.value(statusKey, UserStatus::Status::Online);
```

Comment:
I now realize this could be simplified into:
```cpp
return preDefinedStatus.value(status.toLower(), UserStatus::Status::Online);
```
You don't need the extra line with the ternary operator, a lookup with the empty string will give `UserStatus::Status::Online` anyway thanks to your use of the `value()` extra parameter (good call, I always forget about this one).

Comment just above will need to be dropped or adjusted. If kept I'd just mention why the `toLower()`, the rest is self-explanatory.